src/plugins/saved_search/server/saved_objects/search.ts

```diff
@@ -33,44 +35,74 @@ export function getSavedSearchObjectType(
       },
     },
     mappings: {
+      dynamic: false,
       properties: {
-        columns: { type: 'keyword', index: false, doc_values: false },
         description: { type: 'text' },
-        viewMode: { type: 'keyword', index: false, doc_values: false },
-        hideChart: { type: 'boolean', index: false, doc_values: false },
-        isTextBasedQuery: { type: 'boolean', index: false, doc_values: false },
-        usesAdHocDataView: { type: 'boolean', index: false, doc_values: false },
-        hideAggregatedPreview: { type: 'boolean', index: false, doc_values: false },
-        hits: { type: 'integer', index: false, doc_values: false },
-        kibanaSavedObjectMeta: {
-          properties: {
-            searchSourceJSON: { type: 'text', index: false },
-          },
-        },
-        sort: { type: 'keyword', index: false, doc_values: false },
         title: { type: 'text' },
-        grid: { dynamic: false, properties: {} },
         version: { type: 'integer' },
-        rowHeight: { type: 'text' },
-        timeRestore: { type: 'boolean', index: false, doc_values: false },
-        timeRange: {
-          dynamic: false,
-          properties: {
-            from: { type: 'keyword', index: false, doc_values: false },
-            to: { type: 'keyword', index: false, doc_values: false },
-          },
-        },
-        refreshInterval: {
-          dynamic: false,
-          properties: {
-            pause: { type: 'boolean', index: false, doc_values: false },
-            value: { type: 'integer', index: false, doc_values: false },
-          },
-        },
-        rowsPerPage: { type: 'integer', index: false, doc_values: false },
-        breakdownField: { type: 'text' },
       },
     },
+    schemas: {
+      '8.8.0': schema.object({
+        columns: schema.arrayOf(schema.string(), { defaultValue: [] }),
+        description: schema.string({ defaultValue: '' }),
+        viewMode: schema.maybe(
+          schema.oneOf([
+            schema.literal(VIEW_MODE.DOCUMENT_LEVEL),
+            schema.literal(VIEW_MODE.AGGREGATED_LEVEL),
+          ])
+        ),
+        hideChart: schema.boolean({ defaultValue: false }),
+        isTextBasedQuery: schema.boolean({ defaultValue: false }),
+        usesAdHocDataView: schema.maybe(schema.boolean()),
+        hideAggregatedPreview: schema.maybe(schema.boolean()),
+        hits: schema.maybe(schema.number()),
+        kibanaSavedObjectMeta: schema.object({
+          searchSourceJSON: schema.string(),
+        }),
+        sort: schema.oneOf(
+          [
+            schema.arrayOf(schema.arrayOf(schema.string(), { minSize: 2, maxSize: 2 })),
+            schema.arrayOf(schema.string(), { minSize: 2, maxSize: 2 }),
+          ],
+          { defaultValue: [] }
+        ),
+        title: schema.string({ defaultValue: '' }),
```

Comment:
```suggestion
        title: schema.string(),
```

A title is a required prop, isn't it?

Could you please also move it above `description` line?

Reply:
That's true, I've dropped the `defaultValue` now.

When making these changes, I maintained the original ordering of the `properties`, but I've updated the order now to put `title, description, version` first since they're the only remaining mappings.

Commit: https://github.com/elastic/kibana/pull/154016/commits/b10d13339ec20344ea80f345f8db6887dbef781c.